### ISSUE TITLE
Enable invoicing

### ISSUE THREAD

**Issue by pirapira** [raiden-network/raiden#3767]
## Abstract

In many cases merchants send bills and then customers pay.  This issue keeps track of supporting this use case in Raiden.

## Motivation

@LefterisJP wondered whether this is kept track of.

**Comment by jeemud** [comment]
I think Raiden should only be focus on off chain payments and not specific scenario like merchant payment request and then response...etc, because:

- there are many business scenarios, and Raiden should focus on offchain payments only instead of many BL ( GRASP high cohesion pattern)
-Its easy to develop such business logic on top of raiden by businesses

**Comment by pirapira** [comment]
@jeemud Yes, that sounds like disciplined engineering.  I'll add "Invoicing might not be a part of Raiden.  If Raiden provides enough functionality for others to build invoicing, that's also fine." in the description of this issue.

**Comment by jeemud** [comment]
@pirapira  I agree. I think we can have independent implementations on top of Raiden for merchant usecase. That will be good demo of Raiden also. Are there any set of requirements?

**Comment by pirapira** [comment]
I don't know.  Figuring that out is part of this issue.

**Comment by christianbrb** [comment]
Maybe this is a starting point?
Raiden Invoice Repo https://github.com/ChaeByunghoon/raiden-invoice) made by @chaebyunghoon

**Comment by ChaeByunghoon** [comment]
oh that repo is just made for my demo scenario. That repo just apply concept of bolt11

**Comment by christianbrb** [comment]
Here is a short description of how the Lightning Invoicing works: https://medium.com/suredbits/lightning-101-what-is-a-lightning-invoice-d527db1a77e6

**Comment by drselump14** [comment]
Unlike the original transaction, for the invoice payment the transaction should be initiated by receiver and ask the confirmation from payer by asking the secret.
